### PR TITLE
added trakt provider page

### PR DESCRIPTION
**Pull request by blekmus**
## Changes 💡

Added documentation for the trakt provider

https://github.com/nextauthjs/next-auth/pull/3771

## Screenshot (If Applicable) 📷

Firefox_Screenshot_2022-01-31T11-26-04 606Z

### CODE REVIEW

File: docs/providers/trakt.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+If you're using the api in production by calling [api.trakt.tv](https://api.trakt.tv). Follow the production example. If you wish to develop on Trakt's sandbox environment by calling [api-staging.trakt.tv](https://api-staging.trakt.tv). Follow the development example.
+
+Start by creating an OAuth app on Trakt for [production](https://trakt.tv/oauth/applications/new) or [development](https://staging.trakt.tv/oauth/applications/new). Then set the Client ID and Client Secret as `TRAKT_ID` and `TRAKT_SECRET` in `.env`.
+
+## Options
+
+The **Trakt Provider** comes with a set of default options:
+
+- [Trakt Provider options](/path/to/options/file)
```
Comment (balazsorban44) [inline docs/providers/trakt.md:20]:
```suggestion
- [Trakt Provider options](https://github.com/nextauthjs/next-auth/blob/main/src/providers/trakt.ts)
```

File: docs/providers/trakt.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+You can override any of the options to suit your own use case.
+
+## Example
+
+### Production
+
+~~~js
```
Comment (balazsorban44) [inline docs/providers/trakt.md:28]: could we change the ~~~ to ``` in all places?

File: docs/providers/trakt.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+          { headers: headers }
+        )
+
+        const data = await res.json()
+        return data
+      },
+    },
+  }),
+]
+~~~
```
Comment (balazsorban44) [inline docs/providers/trakt.md:67]:
```suggestion
```

Not sure this section has any value, we link to the default options already. We should remove it IMO.

**Review by balazsorban44 — CHANGES_REQUESTED** [review]

File: docs/providers/trakt.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
+---
+id: trakt
+title: Trakt
+---
+
+## Documentation
+
+https://trakt.docs.apiary.io
```
Comment (balazsorban44) [inline docs/providers/trakt.md:8]:
```suggestion
https://trakt.docs.apiary.io/#reference/authentication-oauth
```

Let's be more specific

File: docs/providers/trakt.md (hunk 3 above)
Comment (blekmus) [reply docs/providers/trakt.md:67]: You've got a point. Alright, I'll remove it

File: docs/providers/trakt.md (hunk 2 above)
Comment (blekmus) [reply docs/providers/trakt.md:28]: Got it

**Comment by blekmus** [comment]
Alright, updated everything with the requested changes.

**Review by balazsorban44 — APPROVED** [review]
Awesome!

**Pull request merged**